admin_manual/installation/nginx.rst

```diff
@@ -114,75 +76,109 @@ webroot of your nginx installation. In this example it is
+      # access it despite the existence of the regex rule `location ~ /\.`
+      # which would otherwise handle requests for `/.well-known`.
+      location ^~ /.well-known {
+          # The following 6 rules are borrowed from `.htaccess`
+      
+          rewrite ^/\.well-known/host-meta\.json  /public.php?service=host-meta-json  last;
+          rewrite ^/\.well-known/host-meta        /public.php?service=host-meta       last;
+          rewrite ^/\.well-known/webfinger        /public.php?service=webfinger       last;
+          rewrite ^/\.well-known/nodeinfo         /public.php?service=nodeinfo        last;
+          
+          location /.well-known/carddav   { return 301 /remote.php/dav/; }
+          location /.well-known/caldav    { return 301 /remote.php/dav/; }
       }
-
-      location ~ ^\/(?:index|remote|public|cron|core\/ajax\/update|status|ocs\/v[12]|updater\/.+|oc[ms]-provider\/.+|.+\/richdocumentscode\/proxy)\.php(?:$|\/) {
-          fastcgi_split_path_info ^(.+?\.php)(\/.*|)$;
+      
+      # Rules borrowed from `.htaccess` to hide certain paths from clients
+      location ~ ^/(build|tests|config|lib|3rdparty|templates|data)($|/)  { return 404; }
+      location ~ ^/(autotest|occ|issue|indie|db_|console)                 { return 404; }
+      
+      # Respond with HTTP 404 to *all* requests for hidden files;
+      # not just hidden files that reside in the root directory.
+      location ~ /\. { return 404; }
+      
+      # Ensure this block, which passes PHP files to the PHP process, is above the blocks
+      # which handle static assets (as seen below). If this block is not declared first,
+      # then Nginx will encounter an infinite rewriting loop when it prepends `/index.php`
+      # to the URI, resulting in a HTTP 500 error response.
+      location ~ \.php($|/) {
+          fastcgi_split_path_info ^(.+?\.php)(/.*|)$;
           set $path_info $fastcgi_path_info;
+          
           try_files $fastcgi_script_name =404;
+          
           include fastcgi_params;
           fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
           fastcgi_param PATH_INFO $path_info;
           fastcgi_param HTTPS on;
-          # Avoid sending the security headers twice
-          fastcgi_param modHeadersAvailable true;
-          # Enable pretty urls
-          fastcgi_param front_controller_active true;
+          
+          fastcgi_param modHeadersAvailable true;         # Avoid sending the security headers twice
+          fastcgi_param front_controller_active true;     # Enable pretty urls
           fastcgi_pass php-handler;
+          
           fastcgi_intercept_errors on;
           fastcgi_request_buffering off;
       }
-
-      location ~ ^\/(?:updater|oc[ms]-provider)(?:$|\/) {
-          try_files $uri/ =404;
-          index index.php;
+      
+      location ~ \.(css|js|svg|gif)$ {
+          try_files $uri $uri/ /index.php$request_uri;
```

Comment:
Do we need the $uri/ here? I don't think we ever need or want file.css to be treated like a folder

Reply:
Not needed, but IMO it simplifies the config's readability if we use the same `try_files` directive everywhere. It won't hurt to remove `$uri/` here.

Reply:
It might improve the consistency of the config, but if you actually read the config logically it doesn't actually make sense.
Closing this since you already implemented the change I recommended.